packages/slate/test/interfaces/Path/transform/move_node/sibling-ends-before-to-ends-equal.tsx

```diff
@@ -9,4 +9,4 @@ const op = {
 export const test = () => {
   return Path.transform(path, op)
 }
-export const output = [0, 0]
+export const output = [0, 1]
```

Comment:
This is wrong. When you move 0 -> 1, Removal of the 0 node causes the 1 to shift forward and take the 0th place. Then what was 0 gets inserted at the newPath which is 1

Example:
Start: [A, B]
Remove 0: [B]
Insert A at 1: [B, A]

B's path has changed to [0, 0]

Suggest you look here: https://github.com/ianstormtaylor/slate/blob/master/packages/slate/src/transforms/general.ts#L91

Reply:
I think what you said is not exactly correct.

The `newPath` [0,1] is defined in the context before `move_node` operation. That means, you can not just insert node at path [0, 1] after remove node at path [0, 0], because the context has already changed. You need to transform the `newPath` [0,1] against `remove_node` operation, use the transformed `newPath` as `insert_node` path.

Example:
Start: [A, B]
Remove 0: [B]
Transform newPath: 1 -> 0
Insert A at 0: [A, B]

The key point is to understand the difference between operation definition context and execution context. Here are some information you can refer to.
[https://www3.ntu.edu.sg/scse/staff/czsun/projects/otfaq/#_Toc321146146](https://www3.ntu.edu.sg/scse/staff/czsun/projects/otfaq/#_Toc321146146)